File: SGPageViewExample/ViewController.m
//
//  ViewController.m
//  SGPageViewExample
//
//  Created by apple on 17/4/12.
//  Copyright © 2017年 Sorgle. All rights reserved.
//

#import "ViewController.h"
#import "DefaultVCOne.h"
#import "DefaultVCTwo.h"
#import "DefaultVCThree.h"
#import "DefaultVCFour.h"
#import "NavigationBarVC.h"
#import "PersonalCenterVC.h"

@interface ViewController () <UITableViewDelegate, UITableViewDataSource>
@property (weak, nonatomic) IBOutlet UITableView *tableView;
@property (nonatomic, strong) NSArray *TitleDataList;
@property (nonatomic, strong) NSArray *VCDataList;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.TitleDataList = @[@"静止样式", @"滚动样式", @"导航栏样式", @"文字渐变效果", @"文字缩放效果", @"微博个人主页"];
    
    DefaultVCOne *oneVC = [[DefaultVCOne alloc] init];
    DefaultVCTwo *twoVC = [[DefaultVCTwo alloc] init];
    NavigationBarVC *navVC = [[NavigationBarVC alloc] init];
    DefaultVCThree *threeVC = [[DefaultVCThree alloc] init];
    DefaultVCFour *fourVC = [[DefaultVCFour alloc] init];
    PersonalCenterVC *PCVC = [[PersonalCenterVC alloc] init];

    self.VCDataList = @[oneVC, twoVC, navVC, threeVC, fourVC, PCVC];
    
    [self foundTableView];
}

- (void)foundTableView {
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cell"];
    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.TitleDataList.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    cell.backgroundColor = [UIColor clearColor];
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    
    cell.textLabel.text = self.TitleDataList[indexPath.row];
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [self.navigationController pushViewController:self.VCDataList[indexPath.row] animated:YES];
}


@end

File: SGPagingView/ViewController.swift
//
//  ViewController.swift
//  SGPagingView
//
//  Created by kingsic on 2021/12/10.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var tableView: UITableView!

    var dataSource = [
        ["固定样式一（回弹效果）": FixedOneVC.self],
        ["固定样式二（动画效果）": FixedTwoVC.self],
        ["滚动样式（指定下标值）": ScrollVC.self],
        ["富文本样式": AttributedVC.self],
        ["文字渐变效果": GradientEffectVC.self],
        ["文字缩放效果": ZoomVC.self],
        ["通知消息样式": BadgeVC.self],
        ["指示器固定样式（爱奇艺首页效果）": IFixedVC.self],
        ["指示器动态样式": IDynamicVC.self],
        ["指示器遮盖样式": CoverOneVC.self],
        ["百度网盘传输界面样式": CoverTwoVC.self],
        ["悬浮效果": SuspensionVC.self],
        ["悬浮效果Pro": SuspensionProVC.self]
    ]

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cellID")
        tableView.tableFooterView = UIView()
    }

}

extension ViewController: UITableViewDataSource {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        dataSource.count
    }
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cellID", for: indexPath)
        cell.accessoryType = .disclosureIndicator
        cell.backgroundColor = .clear
        cell.selectionStyle = .none
        cell.textLabel?.text = dataSource[indexPath.row].keys.first
        return cell
    }
}

extension ViewController: UITableViewDelegate {
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let vc = dataSource[indexPath.row].values.first
        if let tempVC = vc {
            navigationController?.pushViewController(tempVC.init(), animated: true)
        }
    }
}

